packages/client-config/src/generate_client_config_to_file.ts

```diff
@@ -18,20 +19,24 @@ export const generateClientConfigToFile = async (
   outDir?: string,
   format?: ClientConfigFormat,
   // TODO: update this type when Printer interface gets defined in platform-core.
-  log?: (message: string) => void
+  log?: (message: string) => void,
+  version?: number
 ): Promise<void> => {
   const packageJson = await readPackageJson();
 
   const clientConfigWriter = new ClientConfigWriter(
     getClientConfigPath,
-    new ClientConfigFormatter(
-      new ClientConfigConverter(packageJson.name, packageJson.version)
-    )
+    version && version >= 1
```

Comment:
we should explicitly check for known versions and throw on anything unknown / unsupported